docs/tutorial/overview.md

```diff
@@ -10,16 +10,15 @@ prodvec := map 16 (a <- avec, b <- bvec) { a * b }
 dot := reduce 4 (a, b <- prodvec) 0 { a + b }
 ```
 
-We're going to first write a naive compiler with some simplifying assumptions:
+This guide will walk you through the steps to build a program that compiles MrXL programs to FuTIL code. To simplify things, we'll make a few assumptions:
 - Every array in a MrXL program has the same length.
 - Every integer in our generated hardware will be 32 bits.
-- Our generated hardware will not have any parallelism.
-
-Once we've got this simplified compiler working, we can improve it to not require these assumptions any more.
+- Every `map` and `reduce` body will be either a multiplication or addition of either an array element or an integer.
```

Comment:
I think these can go at the end of the tutorial. For example, you could show off the design of the compiler for the one example and pose these restrictions in the current design as challenges that people can try think about.